include/NeoFOAM/timeIntegration/sundials.hpp

```diff
@@ -0,0 +1,375 @@
+ * @param t The current value of the independent variable
+ * @param y The current value of the dependent variable vector
+ * @param ydont The output vector that forms [a portion of] the ODE RHS, f(t, y).
+ * @param user_data he user_data pointer that was passed to ARKodeSetUserData().
+ *
+ * @note https://sundials.readthedocs.io/en/latest/arkode/Usage/User_supplied.html#c.ARKRhsFn
+ *
+ * @details This is our implementation of the RHS of explicit spacital integration, to be integrated
+ * in time. In our case user_data is a unique_ptr to an expression. In this function a 'working
+ * source' vector is created and parsed to the explicitOperation, which should contain the field
+ * variable at the start of the time step. Currently 'multi-stage RK' is not supported until y
+ * can be copied to this field.
+ */
+template<typename SolutionFieldType>
+int explicitRKSolve([[maybe_unused]] sunrealtype t, N_Vector y, N_Vector ydot, void* user_data)
+{
+    // Pointer wrangling
+    NeoFOAM::dsl::Expression* pdeExpre = reinterpret_cast<NeoFOAM::dsl::Expression*>(user_data);
+    sunrealtype* yDotArray = N_VGetArrayPointer(ydot);
+    sunrealtype* yArray = N_VGetArrayPointer(y);
+
+    NF_ASSERT(
+        yDotArray != nullptr && yArray != nullptr && pdeExpre != nullptr,
+        "Failed to dereference pointers in sundails."
+    );
+
+    // Copy initial value from y to source.
+    NeoFOAM::Field<NeoFOAM::scalar> source(pdeExpre->exec(), 1, 0.0);
+    source = pdeExpre->explicitOperation(source); // compute spatial
+    if (std::holds_alternative<NeoFOAM::GPUExecutor>(pdeExpre->exec()))
+    {
+        Kokkos::fence();
+    }
+    NeoFOAM::sundials::fieldToNVector(source, ydot); // assign rhs to ydot.
+    return 0;
+}
+
+// ------------------
+
+namespace detail
+{
+template<typename Vector>
+void initNVector(Vector& vec, size_t size, std::shared_ptr<SUNContext_> context)
+{
+    vec.initNVector(size, context);
+}
+
+template<typename Vector>
+const N_Vector& NVector(const Vector& vec)
+{
+    return vec.NVector();
+}
+
+template<typename Vector>
+N_Vector& NVector(Vector& vec)
+{
+    return vec.NVector();
+}
+}
+
+template<typename ValueType>
+class SKVectorSerial
+{
+public:
+
+    SKVectorSerial() {};
+    ~SKVectorSerial() = default;
+    SKVectorSerial(const SKVectorSerial& other)
+        : kvector_(other.kvector_), svector_(other.kvector_) {};
+
+    SKVectorSerial(SKVectorSerial&& other) noexcept
+        : kvector_(std::move(other.kvector_)), svector_(std::move(other.svector_)) {};
+
+    SKVectorSerial& operator=(const SKVectorSerial& other) = delete;
+    SKVectorSerial& operator=(SKVectorSerial&& other) = delete;
+
+    using KVector = ::sundials::kokkos::Vector<Kokkos::Serial>;
+    void initNVector(size_t size, std::shared_ptr<SUNContext_> context)
+    {
+        kvector_ = KVector(size, context.get());
+        svector_ = kvector_;
+    };
+    const N_Vector& NVector() const { return svector_; };
```

Comment:
because of lower case function
```suggestion
    const N_Vector& getNVector() const { return svector_; };
```